tests/e2e/mapper/bluetooth/bluetooth_test.go

```diff
@@ -0,0 +1,297 @@
+			Token := Client.Subscribe(scheduletopic, 0, WriteDataReceived)
+			if Token.Wait() && TokenClient.Error() != nil {
+				common.Failf("Subscribe to Topic  Failed  %s, %s", TokenClient.Error(), scheduletopic)
+			}
+			Expect(TokenClient.Error()).NotTo(HaveOccurred())
+		})
+		AfterEach(func() {
+			Client.Disconnect(1)
+			common.PrintTestcaseNameandStatus()
+		})
+		It("E2E_READ CHARACTERISTIC FROM WRITTEN DATA: Ensure whether mapper is able to write data and read it back from connected device", func() {
+			//creating deployment from deployment yaml of bluetooth mapper
+			curpath := getpwd()
+			deployPath:="../../../../device/bluetooth_mapper/deployment.yaml"
+			file := path.Join(curpath,deployPath)
+			body, err := ioutil.ReadFile(file)
+			Expect(err).Should(BeNil())
+				client := &http.Client{}
+				BodyBuf := bytes.NewReader(body)
+				req, err := http.NewRequest(http.MethodPost, ctx.Cfg.K8SMasterForKubeEdge+DeploymentHandler, BodyBuf)
+				Expect(err).Should(BeNil())
+				req.Header.Set("Content-Type", "application/yaml")
+				resp, err := client.Do(req)
+				Expect(err).To(BeNil())
+				Expect(resp.StatusCode).Should(Equal(http.StatusCreated))
+			podlist, err := utils.GetPods(ctx.Cfg.K8SMasterForKubeEdge+AppHandler, nodeName)
+			utils.WaitforPodsRunning(ctx.Cfg.K8SMasterForKubeEdge, podlist, 240*time.Second)
+			time.Sleep(30 * time.Second)
+			Eventually(func() bool {
+				return readWrittenData
+			}, "120s", "0.5s").ShouldNot(Equal(false), "Message is not recieved in expected time !!")
+		})
+	})
+
+	// Checking whether dataconverter works correctly
+	Context("Test whether dataconverter works properly", func() {
+		BeforeEach(func() {
+			// Subscribing to topic where scheduler publishes the data
+			ClientOpts = helpers.HubClientInit(ctx.Cfg.MqttEndpoint, "bluetoothmapper", "", "")
+			Client = MQTT.NewClient(ClientOpts)
+			if TokenClient = Client.Connect(); TokenClient.Wait() && TokenClient.Error() != nil {
+				common.Failf("client.Connect() Error is %s", TokenClient.Error())
+			} else {
+				common.Info("Subscribe Connection Successful")
+			}
+			Expect(TokenClient.Error()).NotTo(HaveOccurred())
+			scheduletopic := "$ke/device/bluetooth-mapper/mock-temp-sensor-instance/scheduler/result"
+			Token := Client.Subscribe(scheduletopic, 0, DataConversion)
+			if Token.Wait() && TokenClient.Error() != nil {
+				common.Failf("Subscribe to Topic  Failed  %s, %s", TokenClient.Error(), scheduletopic)
+			}
+			Expect(TokenClient.Error()).NotTo(HaveOccurred())
+			var expectedSchedule []scheduler.Schedule
+			// Create and publish run time data schedule for data conversion.
+			var schedule scheduler.Schedule
+			schedule.Name = "temperatureconversion"
+			schedule.Interval = 3000
+			schedule.OccurrenceLimit = 1
+			schedule.Actions = []string{"ConvertTemperatureData"}
+			expectedSchedule = []scheduler.Schedule{schedule}
+			scheduleCreateTopic1 := "$ke/device/bluetooth-mapper/mock-temp-sensor-instance/scheduler/create"
```

Comment:
please don't suffix with numbers
applicable to similar places

Reply:
Updated